src/shaders/symbol_sdf.vertex.glsl

```diff
@@ -43,8 +50,23 @@ void main() {
     mediump float a_minzoom = a_zoom[0];
     mediump float a_maxzoom = a_zoom[1];
 
-    // u_zoom is the current zoom level adjusted for the change in font size
-    mediump float z = 2.0 - step(a_minzoom, u_zoom) - (1.0 - step(a_maxzoom, u_zoom));
+    float fontScale = u_is_text ? size / 24.0 : size;
+
+    // In order to accommodate placing labels around corners in
+    // symbol-placement: line, each glyph in a label could have multiple layout
+    // "instances", only one of which should be shown at a given zoom level.
+    // The min/max zoom assigned to each instance is based on the font size at
+    // the vector tile's zoom level, which might be different than at the
+    // currently rendered zoom level if text-size is zoom-dependent.
+    // Thus, we compensate for this difference by calculating an adjustment
+    // based on the scale of rendered text size relative to layout text size.
+    mediump float adjustedSize = u_is_size_composite
+        ?  evaluate_zoom_function_1(a_size, u_adjusted_size_t) / 10.0
```

Comment:
It is unfortunate that this breaks the abstraction provided by the #pragma system, and it's also going to need reworking to be usable in native.